### PR TITLE
Addresses #47, #48, and #51

### LINKED ISSUE #47: Add `merge_build_host` transformation
We are missing a transformation of the `merge_build_host` field from CEP-14:
```
  # merge the build and host environments (used in many R packages on Windows)
  # was `merge_build_host`
  merge_build_and_host_envs: bool (defaults to false)
```

`rattler-build` error:
```
Error:   × Parsing: invalid field `merge_build_host`.
    ╭─[50:3]
 49 │   noarch: generic
 50 │   merge_build_host: $${{ true if win }}
    ·   ────────┬───────
    ·           ╰── here
 51 │   rpaths:
    ╰────
```

### PR DESCRIPTION
**Pull request by schuylermartin45**
Fixes #47, #48, and #51 , which are all small changes. After reading through the integration testing logs, I noticed we were missing these easy transformations.

This PR also adds a new transformation function for the `source` section and sorts the `source` section keys.

### CODE REVIEW

**Comment by schuylermartin45** [comment]
Although the integration testing passed, there was a slight dip in the success metrics. ~50 recipes in AnacondaRecipes (and more elsewhere) set `source` to a list of objects:

```
source:
  - path: ./
```
`rattler-build` supports this so I'll need to make a slight adjustment and add a regression test.

**Comment by schuylermartin45** [comment]
If this latest push passes, and the error rate is improved, we should be good to merge.

**Comment by schuylermartin45** [comment]
#50 was a quick win so I've added it to this PR as well.

File: conda_recipe_manager/parser/_types.py [hunk 1]
```diff
@@ -42,6 +42,28 @@
...
+    "git": 50,
+    "branch": 60,
+    "tag": 70,
+    "rev": 80,
+    "depth": 90,
+    "lfs": 100,
+    # Common fields
+    "target_directory": 120,
+    "file_name": 130,
+    "patches": 140,
```
Comment (dholth) [inline conda_recipe_manager/parser/_types.py:64]: This is great like BASIC line numbers :)

File: conda_recipe_manager/parser/recipe_parser_convert.py [hunk 2]
```diff
@@ -196,6 +197,45 @@ def _upgrade_selectors_to_conditionals(self) -> None:
...
+            for src_path in source_paths:
+                # SVN and HG source options are no longer supported. This seems to have been deprecated a long
+                # time ago and there are unlikely any recipes that fall into this camp. Still, we should flag it.
+                if self._new_recipe.contains_value(RecipeParser.append_to_path(src_path, "svn_url")):
+                    self._msg_tbl.add_message(
+                        MessageCategory.WARNING, "SVN packages are no longer supported in the new format"
+                    )
+                if self._new_recipe.contains_value(RecipeParser.append_to_path(src_path, "hg_url")):
+                    self._msg_tbl.add_message(
+                        MessageCategory.WARNING, "HG (Mercury) packages are no longer supported in the new format"
```
Comment (dholth) [inline conda_recipe_manager/parser/recipe_parser_convert.py:229]: 😢 My favorite system. Is there a more permanent name besides "the new format"?
```suggestion
                        MessageCategory.WARNING, "HG (Mercurial) packages are no longer supported in the new format"
```

File: conda_recipe_manager/parser/recipe_parser_convert.py [hunk 3]
```diff
@@ -238,35 +282,16 @@ def _upgrade_build_section(self, base_package_paths: list[str]) -> None:
...
-            "license_url",
-        ]
-        for field in about_required:
-            path = f"/about/{field}"
-            if not self._new_recipe.contains_value(path):
-                self._msg_tbl.add_message(MessageCategory.WARNING, f"Required field missing: {path}")
-
-        # Transform renamed fields
-        about_rename: Final[list[tuple[str, str]]] = [
+        about_rename_mapping: Final[list[tuple[str, str]]] = [
```
Comment (dholth) [inline conda_recipe_manager/parser/recipe_parser_convert.py:290]: How helpful has Final been for you?

**Review by dholth — APPROVED** [review]
Thanks

File: conda_recipe_manager/parser/_types.py (hunk 1 above)
Comment (schuylermartin45) [reply conda_recipe_manager/parser/_types.py:64]: And like BASIC, I've found myself occasionally adding numbers in between lol

File: conda_recipe_manager/parser/recipe_parser_convert.py (hunk 2 above)
Comment (schuylermartin45) [reply conda_recipe_manager/parser/recipe_parser_convert.py:229]: I have "V1" in a few places as that's what the `schema_version` field uses. The "old version" is then referred to as "V0".

But past that, I don't think there is an official nomenclature that anyone has come up with/is using.

File: conda_recipe_manager/parser/recipe_parser_convert.py (hunk 3 above)
Comment (schuylermartin45) [reply conda_recipe_manager/parser/recipe_parser_convert.py:290]: Not as much as I would like. It sucks that you can't use it in a `for` loop. The static analyzer enforces it but it isn't as useful or really does much compared to C++'s const correctness or rust's immutability concepts.

File: conda_recipe_manager/parser/recipe_parser_convert.py (hunk 2 above)
Comment (schuylermartin45) [reply conda_recipe_manager/parser/recipe_parser_convert.py:229]: Also `Mercury` is the name rattler's docs used so I just copied the name from that.

**Comment by dholth** [comment]
https://www.mercurial-scm.org/

**Pull request merged**